### PR TITLE
adds functionality to allow users to see lists of articles for each topic in a separate page, using the navbar links

### PR DESCRIPTION
**Pull request by EliR94**
This includes: 
- adds functionality to allow users to see lists of articles for each topic in a separate page, using the NavBar links
- updates from feedback on PR#7

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ function App() {
       <NavBar />
       <Routes>
         <Route path="/" element={<Articles />}></Route>
+        <Route path="/articles/:topic" element={<Articles />}></Route>
```
Comment (Harrison365) [inline src/App.jsx:15]: nice 👍

File: src/api.js [hunk 2]
```diff
@@ -4,8 +4,16 @@ const newsApi = axios.create({
...
+export const getTopics = () => {
+    return newsApi.get('/topics')
+}
+
+export const getArticles = (topic) => {
+    return newsApi.get('/articles', {
+        params: {
+            topic: topic,
+        }
+    })
```
Comment (Harrison365) [inline src/api.js:16]: Cool 👍

File: src/components/Articles.jsx [hunk 3]
```diff
@@ -1,24 +1,26 @@
...
     
     useEffect(() => {
         setIsLoading(true)
-        getArticles().then((articlesFromAPI) => {
+        getArticles(topic).then((articlesFromAPI) => {
             setArticles(articlesFromAPI.data.articles)
             setIsLoading(false)
         })
-    }, [])
+    }, [topic])
```
Comment (Harrison365) [inline src/components/Articles.jsx:17]: Shouldn't need this in dependancy array.

**Comment by EliR94** [comment]
thank you @Harrison365  🙌

**Pull request merged**